srcalc/src/main/java/gov/va/med/srcalc/web/controller/EnterVariablesController.java

```diff
@@ -135,18 +134,18 @@ public ModelAndView enterVariables(
         catch(final MissingValuesException e)
         {
             // Add all of the missing values to the binding errors
-            for(final MissingValueException missingValue : e.getMissingValues())
+            for(final Variable missingValue : e.getMissingVariables())
             {
                 // If the variable had invalid input, rather than missing input,
                 // it would never be included in the values.
-                final String dynamicKey = VariableEntry.makeDynamicValuePath(missingValue.getVariable().getKey());
+                final String dynamicKey = VariableEntry.makeDynamicValuePath(missingValue.getKey());
                 LOGGER.debug("Field Error: {}", dynamicKey);
                 // Account for field errors on special fields like discrete numerical
                 // variables.
-                if(bindingResultAlreadyContainsError(dynamicKey, valuesBindingResult, missingValue))
+                if (!bindingResultAlreadyContainsError(missingValue, valuesBindingResult))
                 {
-                            valuesBindingResult.rejectValue(
-                                    dynamicKey, ERROR_NO_VALUE, missingValue.getMessage());
+                    valuesBindingResult.rejectValue(
+                            dynamicKey, ERROR_NO_VALUE, "no value");
```

Comment:
I don't think so: it's just a message string for humans, it has no meaning to the code.